get_moochub_courses.php

```diff
@@ -29,190 +32,213 @@
 global $CFG;
 require_once($CFG->libdir . '/filelib.php');
 
+
 $download = optional_param('download', false, PARAM_BOOL);
 
 $metas = [];
-$metarecords = $DB->get_records('ildmeta');
 
 // Create Links for pagination as proposed by the JSON:API schema.
 $jsonlink = $CFG->httpswwwroot . '/local/ildmeta/get_moochub_courses.php';
 $metaslinks = ['self' => $jsonlink, 'first' => $jsonlink, 'last' => $jsonlink];
 $metas['links'] = $metaslinks;
 
-// Get vocabularies from ildmeta_vocabulary for dropdown selection fields.
-$records = $DB->get_records('ildmeta_vocabulary');
-$vocabularies = new \stdClass();
-foreach ($records as $vocabulary) {
-    $vocabularies->{$vocabulary->title} = manager::filter_vocabulary_lang($vocabulary, 'de');
-}
-
-// Get list of providers.
-$providers = manager::get_providers('de');
+$allmetarecs = TRUE;
 
-// Create a json entry for every course, that is supposed to be shared.
-foreach ($metarecords as $meta) {
-    // Skip courses that are not supposed to be exported or a course record does not exist.
-    if ($meta->noindexcourse != 0 || !$DB->record_exists('course', array('id' => $meta->courseid))) {
-        continue;
+// Check if the 'idn' parameter exists.
+if (isset($_GET['idn'])) {
+    $allmetarecs = FALSE;
+    $idnumber = $_GET['idn'];
```

Comment:
Wie heute im Meeting angefragt, ist auch eine Angabe von mehreren idn erwünscht. wir könnte hierzu einen JSON Array nutzen?
```suggestion
    $idnumbers = $_GET['idn'];

    // Check if the input is a valid JSON string.
    if (is_string($idnumbers) and is_array(json_decode($idnumbers, true))) {
        $idnumbers = json_decode($idnumbers, true);
    } else if (!empty($idnumbers)) {
        // Convert to array if only one idnumber is given.
        $idnumbers = [$idnumbers];
    }

    // Check if the input is an array.
    if (empty($idnumbers)) {
        // Send bad request.
        header('HTTP/1.1 400 Bad Request, idn cannot be empty.');
        die();
    }

    // Get all meta records for the selected idnumers.
    [$insql, $inparams] = $DB->get_in_or_equal($idnumbers);
    $sql = "SELECT * FROM {ildmeta} meta
            LEFT JOIN {course} course ON meta.courseid = course.id
            WHERE course.idnumber $insql";
    $metarecords = $DB->get_records_sql($sql, $inparams);
```

Reply:
Habe ich jetzt ganz traditionell über die url parameter gelöst, da ich von Anwenderseite eine Form erwarte, die die Kurse in einer Select list zur Auswahl hätte. Dann wäre keine weitere Schnittstelle und Verrbeitung notwendig. 
Die json Anfrage würde ich eher in einer richtigen REST Schnittstelle unterbringen.